lib/application.js

```diff
@@ -149,6 +149,11 @@ Application.prototype.createClient = function () {
       args.push('spectron-env-' + name + '=' + self.env[name])
     }
 
+    // Port to communicate on. Required starting with ChromeDriver v2.41.
+    // https://bitbucket.org/chromiumembedded/cef/issues/2476/cef-webdriver-instructions-will-need-to
+    const randomPort = Math.floor(Math.random() * (9999 - 9000) + 9000)
```

Comment:
This is not the right port to use, we should be using port `0` and reading the `DevToolsActivePort` file to get the chosen port 🤔

Reply:
From the link above:

> It also doesn't work with CEF. The reason is that CEF's devtools implementation refuses to handle the remote-debugging-port=0 use case. See this code: https://bitbucket.org/chromiumembedded/cef/src/8e9d736cda2015b706e183bfd8f7fd63a2cd5396/libcef/browser/devtools_manager_delegate.cc#lines-75

So I considered that, in Electron (CEF right?), `remote-debugging-port=0` won't work and have not given it a try.

I can if you think I should!

Reply:
@alexstrat I fixed `--remote-debugging-port=0` a while back in Electron -> https://github.com/electron/electron/pull/17800

Reply:
Oh yes! I missed that indeed!

However, it looks like the fix is only available for electron >=5 and this PR is for electron 4.

Should I either:
- backport the fix in Electron 4
- keep this random port solution until Spectron 7 (Electron 5) and add a comment here about it
- try the `remote-debugging-port=0` here and see if it works (is there a reason for it to work?)

Reply:
I started looking into electron 5 and I'm not sure we need to read `DevToolsActivePort` actually. Indeed currently `webdriver@4` does not use the [`devtools-service`](https://github.com/webdriverio-boneyard/wdio-devtools-service) (no trace in dependencies, etc..) which is the only component that'll need access to devtools.

I was able to get it running just by removing `args.push('remote-debugging-port=${randomPort}')`.

Reply:
I chose to keep the random port solution until Spectron 7. #364